site/content/docs/contributing/project-structure.md

```diff
@@ -9,42 +9,139 @@ menu:
 
 
+## Developer Tooling
+Kind includes some tools to help developers maintain the source code compliant to Go best coding practices using tools such as Go fmt, Go lint, and Go vet. It also includes utility scripts that will generate code necessary for kind to make use of Kubernetes-style resource definitions.
+
+Tools are included in the [kind/hack](/kind/hack) directory and fall in one of two categories:
+
+* `update-*` : make changes to the source code
+* `verify-*` : verify that source code is in a good state
+
+We will proceed by describing all of the current tooling in [/kind/hack](/kind/hack).
+
+### Update
+In order to get the project’s source code into a state likely to be compilable, the script
+[update-all.sh](/kind/hack/update-all.sh) performs the following tasks:
+
+* runs update-deps.sh to get obtain all of the project’s dependencies
+* runs update-generated.sh to generate code necessary to define a configuration schema for kind, see https://kind.sigs.k8s.io/docs/user/quick-start/#configuring-your-kind-cluster
+* runs update-gofmt.sh which formats all Go source code using the Go fmt tool.
+
+Let’s go a little in depth on each of these files.
+
+[update-deps.sh](/kind/hack/update-deps.sh) performs the following steps:
+
+* runs `go mod tidy` to rune any no-longer-needed dependencies from go.mod and add any dependencies needed for other combinations of OS, architecture, and build tags
+* runs `go mod vendor`, which creates a vendor directory, resets the main module's vendor directory to include all packages needed to build and test all of the module's packages based on the state of the go.mod files and Go source code.
+* finally, it prunes the vendor directory of any unnecessary files (keeps code source files, licenses, author files, etc).
+
+[update-gofmt.sh](/kind/hack/update-gofmt.sh) runs gofmt on all nonvendored source code to format and simplify the code.
+
+
+[update-generated.sh](/kind/hack/update-generated.sh) in short, generates Go source code that is necessary to use a Kubernetes-style resource definition to define a schema that can be use to configure Kind.
+
+Going a bit more in depth, update-generated.sh Installs
+[go-bindata](https://github.com/jteeuwen/go-bindata)
+which is used to embed static assets in Go. It also installs the deepcopy-gen, defaulter-gen, and conversion-gen tools from
+[kubernetes/code-generator](https://github.com/kubernetes/code-generator).
+These programs  are used to generate Kubernetes-like APIs. These programs are run in the following sequence: deepcopy -> defaulter -> conversion.
+
+To understand this process better, we need to keep in mind that kind’s configuration schema dictates how to bootstrap a Kubernetes cluster. The schema is defined in
+[kind/pkg/cluster/config](/pkg/cluster/config).
+In this directory, currently, you will see two subdirectories:
+[v1beta1](/pkg/cluster/config/v1beta1) and
+[v1beta2](/pkg/cluster/config/v1beta2).
```

Comment:
It might be better to use absolute URLs to specific revisions here, so that we don't accidentally break the links in future? Not sure 🤔

Reply:
fixed